cli/runtime/apis/cli/v1alpha1/catalog_types.go

```diff
@@ -158,6 +158,17 @@ type PluginDescriptor struct {
 	PostInstallHook Hook `json:"-" yaml:"-"`
 }
 
+// PluginInfo describes a plugin information. This is a super set of PluginDescriptor
+// It includes some additional metadata that plugin runtime configures
+type PluginInfo struct {
+	// PluginDescriptor describes a plugin binary.
+	PluginDescriptor `json:",inline" yaml:",inline"`
+
+	// PluginRuntimeVersion of the plugin. Must be a valid semantic version https://semver.org/
+	// This version specifies the version of Plugin Runtime that was used to build the plugin
+	PluginRuntimeVersion string `json:"pluginRuntimeVersion" yaml:"pluginRuntimeVersion"`
```

Comment:
This is very elegant in itself.  However, looking at it in detail now, I wonder why we need it at all?  It is artificially used to provide a json-formatted output for the `info` command.  In such a case, couldn't it be a local, private type in the `info.go` file?

What I mean is that we don't actually pass any info through this type.  We just store it and print it right away...

Reply:
Makes sense. The comment make sense. I was thinking a bit more futuristic when Core CLI need to parse the output of the `tanzu <plugin> info` command and read the `PluginRuntimeVersion` version to take some decision. We are not there yet, but might be there soon. At that time we might need this to be a public API.

Considering that, Should we keep this private/local type for now and when needed make it public? (I am fine with keeping this private for now considering no-one is consuming it)

Reply:
You are right that we will probably need it in the future.
Personally, the projects I worked on before had to 100% respect their public API (like Cobra for example), so my instinct to always avoid creating new APIs until they are absolutely needed, so as to avoid any risk of being stuck with them.

I don't know if that mentality applies to us yet, but IIUC it will in the future, so I personally would avoid any unnecessary new APIs.  But I leave the decision to you and @vuil

Reply:
Sure. I can make this private and when needed in future we can make it public as per the requirements. This make more sense to me as well.